File: src/main/java/tn/esprit/foyer/Entity/Chambre.java
package tn.esprit.foyer.Entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Chambre implements Serializable {
    public enum TypeChambre { SIMPLE, DOUBLE, TRIPLE };
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long idChambre;
    private long numeroChambre;
    @Enumerated(EnumType.STRING)

    private TypeChambre typeChambre;
    @ManyToOne
    @JsonBackReference
    Bloc bloc;
    @OneToMany(cascade = CascadeType.ALL, mappedBy="chambre")
    private Set<Reservation> Reservations;
}

File: src/app/models/Chambre.ts
import { Bloc } from '../models/Bloc';
import { Reservation } from './Reservation';

export enum TypeChambre {
    SIMPLE = 'SIMPLE',
    DOUBLE = 'DOUBLE',
    TRIPLE = 'TRIPLE'
}

export class Chambre {
    idChambre: number;
    numeroChambre: number;
    typeChambre: TypeChambre;
    bloc?: Bloc;
    reservations?: Reservation[];

    constructor(
        idChambre: number,
        numeroChambre: number,
        typeChambre: TypeChambre,
        bloc?: Bloc,
        reservations?: Reservation[]
    ) {
        this.idChambre = idChambre;
        this.numeroChambre = numeroChambre;
        this.typeChambre = typeChambre;
        this.bloc = bloc;
        this.reservations = reservations;
    }
}
